src/modules/computerVision/gaborFilter/gaborFilter.tsx

```diff
@@ -0,0 +1,184 @@
+      //   const K = Math.PI;
+        
+        const gabor = gaborFilter(kernelSize, omega, theta, func, K)
+        const newKernel = gabor.flat();
+        const newKernelGrid = gabor;
+        setKernel(newKernel);
+        setKernelGrid(newKernelGrid);
+    }
+
+    return (
+        <div className="m-4">
+            <KernelConfig onConfig={configureKernel}/>
+            <div className="m-4">
+                Gaussian Kernel
+                <table className="m-auto"><tbody>
+                    {kernelGrid?.map((row, i) => (
+                        <tr key={i}>
+                            {row.map((val, j) => (
+                                <td key={j}
+                                    style={getBg(val, kernel)}
+                                    className="border-2 border-charcoal p-2"
+                                    title={`${val}`}>
+                                    {val.toFixed(5)}
+                                </td>))}
+                        </tr>
+                    ))}
+                </tbody></table>
+            </div>
+            <InteractiveFilter kernel={kernel} imgUrl={jellyfish} />
+        </div>
+    )
+}
+
+const KernelConfig = (props: { onConfig: (kernelSize: number, omega: number, theta: number, K: number) => void }) => {
+    const [kernelSize, setKernelSize] = useState<number>(5);
+    const [omega, setOmega] = useState<number>(1);
+    const [theta, setTheta] = useState<number>(0);
+    const [K, setK] = useState<number>(3.14);
+
+    const changeOmega = (e: any) => setOmega(parseFloat(e.target.value));
+    const changeKernelSize = (e: any) => setKernelSize(parseInt(e.target.value));
+    const changeTheta = (e: any) => setTheta(parseFloat(e.target.value));
+    const changeK = (e: any) => setK(parseFloat(e.target.value));
+
+    const invalidSize = (kernelSize < 3 || kernelSize > 101)
+    const invalidConfig = invalidSize;
+    return (
+        <div>
+            <div className="font-bold m-3">
+                Omega
+                <input className="mx-2 w-64"
+                    type="range" min="0" max="10"
+                    value={omega} onChange={(e) => changeOmega(e)} />
+                <input className="number-input"
+                    type="number" min="0" max="10"
+                    value={omega} onChange={(e) => changeOmega(e)} />
+            </div>
+            <div className="font-bold m-3 h-10">
+                Kernel Size
+                <input className="mx-2 w-64"
+                    type="range" min="3" max="101" step={2}
+                    value={kernelSize} onChange={(e) => changeKernelSize(e)} />
+                <input className="number-input"
+                    type="number" min="3" max="101" step={2}
+                    value={kernelSize} onChange={(e) => changeKernelSize(e)} />
+                <div className="font-light italic text-sm">
+                    { invalidSize ? 'Enter an odd kernel size, between 3 and 101' : ''}
```

Comment:
Gabor does not need an odd kernel size, right? This error message can probably be updated.